File: Easy/0704 - binarysearch/binaryseach.js
var search = function(nums, target)
{
    let start = 0;   
    let end = nums.length;

    while (start < end)
    {
        let median = Math.floor(start + (end - start) / 2);
        if (target < nums[median])
            end = median;
        else if (target > nums[median])
            start = median + 1;
        else
            return median;
    }

    return -1;
};

File: Easy/0704 - binarysearch/binaryseach.py
class Solution:
    def search(self, nums: list[int], target: int) -> int:
        start = 0;   
        end = len(nums)

        while (start < end) :
            median = math.floor(start + (end - start) / 2)
            if (target < nums[median]) :
                end = median
            elif (target > nums[median]) :
                start = median + 1
            else :
                return median

        return -1